tests/components/homekit_controller/common.py

```diff
@@ -227,6 +227,30 @@ def add_characteristic(self, name):
     return pairing
 
 
+async def device_config_changed(hass, accessories):
+    """Discover new devices added to HomeAssistant at runtime."""
+    # Update the accessories our FakePairing knows about
+    controller = hass.data[CONTROLLER]
+    pairing = controller.pairings['00:00:00:00:00:00']
+    pairing.accessories = accessories
+
+    discovery_info = {
+        'host': '127.0.0.1',
+        'port': 8080,
+        'properties': {
+            'md': 'TestDevice',
+            'id': '00:00:00:00:00:00',
+            'c#': '2',
+            'sf': '0',
+        }
+    }
+
+    fire_service_discovered(hass, SERVICE_HOMEKIT, discovery_info)
```

Comment:
Although this is a test, we should still not use the sync api from async context.

Reply:
How do you want to proceed? This will be removed by next PR anyway, although I had planned to wait a cycle to avoid doing too much at once.

Reply:
Call `hass.bus.async_fire` directly instead of using the test helper. Or add an async version of this helper.